### PR TITLE
Implements #28 Support claim "preferred_username" as username for generic oidc and Issuer & Subject

### PR DESCRIPTION
**Pull request by droid42**
* implement unvalidated parsing of token for "whoami" with all possible fields including Issuer & Subject
* prefer "perferred_username" over "name" if filled

### CODE REVIEW

**Comment by droid42** [comment]
Wait for https://github.com/metal-stack/security/pull/18 and tag

File: auth/auth.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ import (
 const cloudContext = "cloudctl"
 
 var DexScopes = []string{"groups", "openid", "profile", "email", "federated:id"}
-var GenericScopes = []string{"groups", "openid", "profile", "email"}
+var GenericScopes = []string{"openid", "profile", "email"}
```
Comment (michaelottofits) [inline auth/auth.go:32]: ok

File: auth/auth.go [hunk 2]
```diff
@@ -108,16 +108,30 @@ func (a *app) Consolef(format string, args ...interface{}) {
...
-	Name          string      `json:"name"`
+type Claims struct {
+	Id              string            `json:"jti,omitempty"`
+	ExpiresAt       int64             `json:"exp,omitempty"`
+	IssuedAt        int64             `json:"iat,omitempty"`
+	NotBefore       int64             `json:"nbf,omitempty"`
+	Issuer          string            `json:"iss,omitempty"`
+	Subject         string            `json:"sub,omitempty"`
+	Audience        interface{}       `json:"aud,omitempty"`
+	Groups          []string          `json:"groups"`
```
Comment (michaelottofits) [inline auth/auth.go:119]: is groups here needed ?

**Review by michaelottofits — APPROVED** [review]
looks good

File: auth/auth.go (hunk 2 above)
Comment (droid42) [reply auth/auth.go:119]: yes, this struct contains all possible claims for dex and generic tokens

**Pull request merged**